### ISSUE TITLE
Setting cloudwatch log group so that logs appear in Cloudwatch Traces

### ISSUE THREAD

**Issue by nikhil-pandey** [aws-observability/aws-otel-collector#1766]
**Describe the question**
We have multiple services in ECS, aws-otel-collector is one of them and all the services report traces to this one instance of aws-otel-collector. Right now only traces data go through the aws-otel-collector.

Here's the config we're using
```yaml
extensions:
  health_check:

receivers:
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
      http:
        endpoint: 0.0.0.0:4318

processors:
  batch/traces:
    timeout: 1s
    send_batch_size: 50
  batch/metrics:
    timeout: 60s

exporters:
  awsxray:
    index_all_attributes: true

service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch/traces]
      exporters: [awsxray]

  extensions: [health_check]
```

Each service has two environment variables set: `OTEL_RESOURCE_ATTRIBUTES` and `OTEL_SERVICE_NAME`. `OTEL_RESOURCE_ATTRIBUTES` contains the following:
- `service.namespace`
- `service.version`
- `container.image.tag`
- `aws.log.group.names`
- `aws.log.group.arns`

Here's what the xray document looks like in cloudwatch
```
{
    "Id": "1-63c8b8cc-5e6c41bad8225f90bc5bc3c0",
    "Duration": 0.008,
    "LimitExceeded": false,
    "Segments": [
        {
            "Id": "5c2e8e8f2b2197cd",
            "Document": {
                "id": "5c2e8e8f2b2197cd",
                "name": "AppName",
                "start_time": 1674098892.2819598,
                "trace_id": "1-63c8b8cc-5e6c41bad8225f90bc5bc3c0",
                "end_time": 1674098892.2904406,
                "fault": false,
                "error": false,
                "throttle": false,
                "http": {
                    "request": {
                        "url": "http://<Url>",
                        "method": "POST",
                        "user_agent": "<UserAgent>"
                    },
                    "response": {
                        "status": 200,
                        "content_length": 0
                    }
                },
                "aws": {
                    "ecs": {
                        "container_id": "c4ae1a0b08671a6312fa7/e8538e8f62bc4ae1a0b08671a6312fa7-265927825"
                    },
                    "xray": {
                        "auto_instrumentation": false
                    }
                },
                "annotations": {
                    "otel_resource_cloud_provider": "aws",
                    "otel_resource_container_runtime": "docker",
                    "http_route": "/some/fun/route",
                    "http_flavor": "1.1",
                    "otel_resource_cloud_platform": "aws_ecs",
                    "otel_resource_container_image_name": "<accountId>.dkr.ecr.<region>.amazonaws.com/<repo>",
                    "otel_resource_container_image_tag": "AmazingTag",
                    "otel_resource_container_id": "c4ae1a0b08671a6312fa7/e8538e8f62bc4ae1a0b08671a6312fa7-265927825",
                    "otel_resource_aws_log_group_names": "MyAmazingLogGroup/Development",
                    "otel_resource_service_version": "5ba787d34761bcf4486b79eec9e4666486f79795",
                    "otel_resource_aws_log_group_arns": "arn:aws:logs:<region>:<accountId>:log-group:MyAmazingLogGroup/Development:*",
                    "otel_resource_service_namespace": "Development",
                    "otel_resource_service_name": "AppName"
                },
                "service": {
                    "version": "5ba787d34761bcf4486b79eec9e4666486f79795"
                },
                "origin": "AWS::ECS::Container",
                "subsegments": [
                    ...
                ]
            }
        },
       ...
    ]
}
```

**What did you expect to see?**

Logs don't show up on Traces page.

- Setting `aws.log.group.arns` metadata does nothing?
- Based on [this doc](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html), `aws.cloudwatch_logs.log_group` and `aws.cloudwatch_logs.arn` define the log group in Segment document. Will setting that fix this? Is there a way to set this or plans to get this from opentelemetry schema?

![Screenshot 2023-01-18 at 10 36 22 PM](https://user-images.githubusercontent.com/3397134/213349600-0a27998e-89a1-44e5-a759-abc3a9cd394f.png)

**Comment by rapphil** [comment]
`aws.log.group.names` and `aws.log.group.arns` are arrays and currently can only be set programmatically. It is not possible to set them using `OTEL_RESOURCE_ATTRIBUTES`. 

Having said that, a solution for your issue will be available in the next collector release v0.26.0 when the following features from the awsxray exporter are available: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsxrayexporter#traces-and-logs-correlation

**Comment by nikhil-pandey** [comment]
I tried the programatic way. We're using C#. Added tags to activity.
Shows up nicely on the console
![Screenshot 2023-01-19 at 1 29 50 AM](https://user-images.githubusercontent.com/3397134/213371732-a5252f78-233e-4b96-9fcd-c800fba4698a.png)

I enabled logging on the collector and this is what i see
![Screenshot 2023-01-19 at 1 38 58 AM](https://user-images.githubusercontent.com/3397134/213373126-ba74f752-67b0-43dc-81cd-37f59043eba3.png)


`aws.log.group.names` and `aws.log.group.arns` stopped showing on raw trace document in console. 

And I cant see any logs 
![Screenshot 2023-01-19 at 1 33 15 AM](https://user-images.githubusercontent.com/3397134/213372079-b35e831b-459d-4873-a8c8-1f5f51685287.png)

**Comment by nikhil-pandey** [comment]
The query looks right with the right trace. Log group however does not get selected automatically? 

![Screenshot 2023-01-19 at 7 05 11 PM](https://user-images.githubusercontent.com/3397134/213589161-6c54ec02-b1a1-4085-8d7b-94e34ca0219a.png)

**Comment by rapphil** [comment]
It seems that you did everything right. Can you just check for me what is the raw content of your trace? Select your trace in the Xray console and then click in "Raw Data".

In the sample application that I'm running, mine looks like this:
```
{
    "Id": "1-63ca25bc-71c597f94e9cc7a4204fc13b",
    "Duration": 0.003,
    "LimitExceeded": false,
    "Segments": [
        {
            "Id": "e4bb6a8d19772711",
            "Document": {
                "id": "e4bb6a8d19772711",
                "name": "java-sample-app",
                "start_time": 1674192316.524301,
                "trace_id": "1-63ca25bc-71c597f94e9cc7a4204fc13b",
                "end_time": 1674192316.5273387,
                "fault": false,
                "error": false,
                "throttle": false,
                "http": {
                    "request": {
                        "url": "http://localhost:4567/",
                        "method": "GET"
                    },
                    "response": {
                        "status": 200,
                        "content_length": 0
                    }
                },
                "aws": {
                    "xray": {
                        "auto_instrumentation": false,
                        "sdk_version": "1.21.0",
                        "sdk": "opentelemetry for java"
                    },
                    "cloudwatch_logs": [
                        {
                            "log_group": "sample-app-trace-logs"
                        }
                    ]
                },
                "metadata": {
                    "default": {
                        "otel.resource.telemetry.sdk.name": "opentelemetry",
                        "otel.resource.aws.log.group.names": [
                            "sample-app-trace-logs"
                        ],
                        "otel.resource.service.name": "java-sample-app",
                        "otel.resource.telemetry.sdk.language": "java",
                        "otel.resource.telemetry.sdk.version": "1.21.0"
                    }
                }
            }
        }
    ]
}
```

Can you check if you have something similar in the AWS section?

```
                    "cloudwatch_logs": [
                        {
                            "log_group": "sample-app-trace-logs"
                        }
                    ]
```

**Comment by nikhil-pandey** [comment]
Its missing cloudwatch_logs section
```
                "aws": {
                    "ecs": {
                        "container_id": "b4112add3a15ea8f2bf7c/5d57e9db170b4112add3a15ea8f2bf7c-265927825"
                    },
                    "xray": {
                        "auto_instrumentation": false
                    }
                },
```

Logs from otel collector show `aws.log.group.names` and `aws.log.group.arns` but are missing from the raw trace document. The collector is not mapping them correctly / they get dropped?

![](https://user-images.githubusercontent.com/3397134/213373126-ba74f752-67b0-43dc-81cd-37f59043eba3.png)

**Comment by nikhil-pandey** [comment]
Or do i have to emit them as `aws.cloudwatch_logs.log_group` instead? Isnt collector supposed to do the mapping?

**Comment by rapphil** [comment]
To be clear, you are setting the attributes as resource attributes right? not attributes of the trace itself. I wonder why the metadata section of your original trace is missing here https://github.com/aws-observability/aws-otel-collector/issues/1766#issue-1548467804.

**Comment by nikhil-pandey** [comment]
Oh, metadata is missing due to`index_all_attributes: true`. Which apparently converts all metadata to `annotations` where `.` is replaced with `_`. Even this, we did notice any difference with the behavior.

**Comment by rapphil** [comment]
I see. were you able to test again?

**Comment by nikhil-pandey** [comment]
Yes. Its working now after using adding it programmatically to the metadata as a list
This is what the aws section looks like now
```json
                "aws": {
                    "ecs": {
                        "container_id": "5442885a8b5125b858fe2/7f9b08ba1805442885a8b5125b858fe2-265927825"
                    },
                    "xray": {
                        "auto_instrumentation": false,
                        "sdk_version": "1.4.0.753",
                        "sdk": "opentelemetry for dotnet"
                    },
                    "cloudwatch_logs": [
                        {
                            "log_group": "MyService/Environment",
                            "arn": "arn:aws:logs:us-west-2:ACCOUNTID:log-group:MyService/Environment:*"
                        }
                    ]
                },
```

**Comment by nikhil-pandey** [comment]
Clicking on "View CloudWatch logs Insights" button does take me to insights page with the log group pre-populated and clicking on "Run Query" shows me logs as expected
![Screenshot 2023-01-23 at 2 56 56 PM](https://user-images.githubusercontent.com/3397134/214137092-2cf410f0-b5ff-4cc9-8090-1f31560081af.png)

But Logs section on the Trace Page is still empty
![Screenshot 2023-01-23 at 2 58 16 PM](https://user-images.githubusercontent.com/3397134/214137388-96a2759a-524a-4b86-ac85-46bc07b6ab15.png)

**Comment by rapphil** [comment]
you need to inject the trace id in the logs for everything to work end end. How this is done is Language specific. For Java, we provide support to Log4j and Logback. Both are able to do this using MCD.

Under the hood what happens is that for every logged message, the trace id string is included.

https://github.com/aws-observability/aws-otel-community/blob/master/sample-apps/java-sample-app/manual/src/main/resources/log4j2.xml#L6
https://github.com/aws-observability/aws-otel-community/blob/master/sample-apps/java-sample-app/manual/src/main/java/software/amazon/adot/sampleapp/AwsXrayContextDataProvider.java#L12


I'm not sure what is the proper mechanism for DotNet.

**Comment by nikhil-pandey** [comment]
We already have. The insights page just works. Trace page doesnt.

![Screenshot 2023-01-23 at 3 06 33 PM](https://user-images.githubusercontent.com/3397134/214138945-76343d9f-8776-44d9-a117-7696b9c6b54f.png)

**Comment by rapphil** [comment]
Can you try the following? from a trace id that is know to be in the logs, find the corresponding trace in the xray console?

in your example, can you try to find the trace `1-63cee...`in the xray trace id? It should appear in the bottom part.

**Comment by nikhil-pandey** [comment]
That works. Xray console doesnt show any logs though. You have to go to insights and run the query manually to find the logs.

![Screenshot 2023-01-23 at 4 50 48 PM](https://user-images.githubusercontent.com/3397134/214157834-7ea7c771-dbbf-4a47-b8ae-cc9ec93b7b8c.png)
![Screenshot 2023-01-23 at 4 51 09 PM](https://user-images.githubusercontent.com/3397134/214157883-e9e2633e-d6aa-4a83-9b64-659245928c2f.png)

**Comment by rapphil** [comment]
Did you have to change the time range for the log to appear in logs insight?

**Comment by nikhil-pandey** [comment]
I dont have to change the time in insights. Just click run.

**Comment by rapphil** [comment]
@willarmiros do you have any suggestion of how we can proceed from here?

**Comment by advanderveer** [comment]
I've been struggling with the same problem (otel collector in separate service, on ECS) in the past few days. My raw traces look like this:
```
{
    "Id": "1-63d77907-eaaecd79d389981b36c0e1f6",
    "Duration": 0.111,
    "LimitExceeded": false,
    "Segments": [
        {
            "Id": "18ed716421b53c34",
            "Document": {
                "id": "18ed716421b53c34",
                "name": "<SOME SERVICE>",
                "start_time": 1675065607.2129626,
                "trace_id": "1-63d77907-eaaecd79d389981b36c0e1f6",
                "end_time": 1675065607.3237371,
                "fault": false,
                "error": false,
                "throttle": false,
                "http": {
                    "request": {
                        "url": "<SOMEURL>",
                        "method": "GET",
                        "user_agent": "<SOME AGENT>",
                        "client_ip": "95.97.200.46",
                        "x_forwarded_for": true
                    },
                    "response": {
                        "status": 200,
                        "content_length": 0
                    }
                },
                "aws": {
                    "ecs": {
                        "container": "87e6c5c56a0c",
                        "task_arn": "arn:aws:ecs:eu-west-1:<ACCOUNTID>:task/ClPlatform-ClusterMain28426E18-yyjktFpDnevD/0b2ee3b584354df3af60d2d38fe27024",
                        "cluster_arn": "arn:aws:ecs:eu-west-1:<ACCOUNTID>:cluster/ClPlatform-ClusterMain28426E18-yyjktFpDnevD",
                        "task_family": "ClBackend999GraphApiTaskDefinitionA86D9462",
                        "container_id": "87e6c5c56a0ceaad433e981d81620fc84f74c534728c75ed6b791122ac109934",
                        "container_arn": "arn:aws:ecs:eu-west-1:<ACCOUNTID>:container/ClPlatform-ClusterMain28426E18-yyjktFpDnevD/0b2ee3b584354df3af60d2d38fe27024/84c45d55-0803-47b8-a45b-8c72324202e4",
                        "launch_type": "ec2"
                    },
                    "xray": {
                        "auto_instrumentation": false
                    }
                },
                "metadata": {
                    "default": {
                        "otel.resource.aws.ecs.task.arn": "arn:aws:ecs:eu-west-1:<ACCOUNTID>:task/ClPlatform-ClusterMain28426E18-yyjktFpDnevD/0b2ee3b584354df3af60d2d38fe27024",
                        "net.transport": "ip_tcp",
                        "otel.resource.aws.log.stream.arns": "arn:aws:logs:eu-west-1:<ACCOUNTID>:log-group:ClBackend999-GraphApiTaskDefinitiongraphapiLogGroupCA1D37A2-Aj4dDxk49XCX:log-stream:graphapi/graphapi/0b2ee3b584354df3af60d2d38fe27024",
                        "http.flavor": "1.1",
                        "otel.resource.aws.log.group.names": "ClBackend999-GraphApiTaskDefinitiongraphapiLogGroupCA1D37A2-Aj4dDxk49XCX",
                        "otel.resource.aws.ecs.cluster.arn": "arn:aws:ecs:eu-west-1:<ACCOUNTID>:cluster/ClPlatform-ClusterMain28426E18-yyjktFpDnevD",
                        "otel.resource.aws.ecs.task.family": "ClBackend999GraphApiTaskDefinitionA86D9462",
                        "otel.resource.aws.ecs.launchtype": "ec2",
                        "otel.resource.aws.ecs.task.revision": "38",
                        "otel.resource.aws.log.group.arns": "arn:aws:logs:eu-west-1:<ACCOUNTID>:log-group:ClBackend999-GraphApiTaskDefinitiongraphapiLogGroupCA1D37A2-Aj4dDxk49XCX:*",
                        "otel.resource.container.id": "87e6c5c56a0ceaad433e981d81620fc84f74c534728c75ed6b791122ac109934",
                        "otel.resource.aws.ecs.container.arn": "arn:aws:ecs:eu-west-1:<ACCOUNTID>:container/ClPlatform-ClusterMain28426E18-yyjktFpDnevD/0b2ee3b584354df3af60d2d38fe27024/84c45d55-0803-47b8-a45b-8c72324202e4",
                        "http.route": "/*",
                        "otel.resource.cloud.platform": "aws_ecs",
                        "otel.resource.aws.log.stream.names": "graphapi/graphapi/0b2ee3b584354df3af60d2d38fe27024",
                        "otel.resource.cloud.provider": "aws",
                        "otel.resource.container.name": "87e6c5c56a0c"
                    }
                },
                "origin": "AWS::ECS::EC2",
                "subsegments": [] ...
```

As you can see, it sets the "aws.log.group.arns" and "aws.log.group.names" as the document suggests. But it doesn't create the "cloudwatch_log" section for xray. The otel resource attributes are set go sdk's ECS resource detector: https://pkg.go.dev/go.opentelemetry.io/contrib/detectors/aws/ecs (the latest version. 

Like is described in this issue, it does NOT set these attributes as string slices, but as JUST a string. When I create my own resource detector and set these attributes as string slices instead, the trace to log correlation starts to work and the collector adds the "cloudwatch_log" section for xray.

The documentation for the xray exporter: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsxrayexporter#exporter-configuration seems to suggest it SHOULD accept string values for these attributes. But it does not (anymore?).

Either the ECS resource detector needs to be upgraded or the collector code needs to accept string values for these attributes instead of just accepting explicitly string slices.

Hopefully this helps

**Comment by advanderveer** [comment]
Actually, even with all this effort the only thing I'm getting functional is that the log group is automatically selected when I press "view in cloudwatch log insights". There it will find logs for a trace after a few minutes. but they never appear in the xray trace interface (even with time range configured for going back hours).

**Comment by willarmiros** [comment]
> Either the ECS resource detector needs to be upgraded or the collector code needs to accept string values for these attributes instead of just accepting explicitly string slices.

I think the action should be on us to update the X-Ray exporter. Technically the spec requires `aws.log.group.names` to be a slice, but we're seeing an increased number of use cases where setting it as string is just more sensible, so the X-Ray exporter should translate both. @rapphil since you are working on the `OTEL_RESOURCE` env var support for this attribute to be a string, do you think you could also make this update in the exporter logic?

> but they never appear in the xray trace interface (even with time range configured for going back hours).

@advanderveer the reason is likely because you are not injecting your trace ID into logs. This is something you'd need to configure in your OTel Go SDK, but I'm not familiar with its level of support for trace ID injection, are you aware of any support?

**Comment by nikhil-pandey** [comment]
String support was already added by @rapphil, it will be shipped with the next version

> but they never appear in the xray trace interface (even with time range configured for going back hours).

@willarmiros If you check my screenshots above, trace id injection is done properly. CouldWatch Insights shows me the logs. But Xray doesnt.

**Comment by willarmiros** [comment]
Thanks for clarifying @nikhil-pandey - I will forward this to the team that owns this console view and try to understand why the logs may not be auto-populating

**Comment by nikhil-pandey** [comment]
One more thing we noticed with *log group not getting selected automatically in cloudwatch*

After making it a slice, in a single account setup, the log group gets pre-selected when you go to cloudwatch logs from xray.

But in a multi-account setup (with a few source accounts and a monitoring account, using the monitoring account for query), it doesn't work. Is there any additional field missing in the trace document? Its only happening to traces coming from the collector, traces instrumented directly using xray sdk work perfectly with this setup.

**Comment by advanderveer** [comment]
> > Either the ECS resource detector needs to be upgraded or the collector code needs to accept string values for these attributes instead of just accepting explicitly string slices.
> 
> I think the action should be on us to update the X-Ray exporter. Technically the spec requires `aws.log.group.names` to be a slice, but we're seeing an increased number of use cases where setting it as string is just more sensible, so the X-Ray exporter should translate both. @rapphil since you are working on the `OTEL_RESOURCE` env var support for this attribute to be a string, do you think you could also make this update in the exporter logic?
> 
> > but they never appear in the xray trace interface (even with time range configured for going back hours).
> 
> @advanderveer the reason is likely because you are not injecting your trace ID into logs. This is something you'd need to configure in your OTel Go SDK, but I'm not familiar with its level of support for trace ID injection, are you aware of any support?

I am injecting the trace id in the logs. That is why the logs DO show up when I press "view in cloudwatch log insights" (and press "run query") in the xray trace interface:


https://user-images.githubusercontent.com/184876/215758925-206282b7-d274-4ca7-88ae-d29f828ac421.mov

**Comment by mderriey** [comment]
> I am injecting the trace id in the logs

_Take the below with a pinch of salt given I haven't been able to make this work yet, I stumbled here while doing some research_

My understanding is that it's not just the trace id that needs to appear in the logs, but also the span (or segment in X-Ray terminology) id, with the format `<trace-id>@<span-id>`.

Sources for this:

- Docs: https://docs.aws.amazon.com/xray/latest/devguide/xray-sdk-java-configuration.html#xray-sdk-java-configuration-logging:~:text=Trace%20ID%20injection%20into%20logs
- Java implementation for log4j: <https://github.com/aws-observability/aws-otel-community/blob/master/sample-apps/java-sample-app/manual/src/main/java/software/amazon/adot/sampleapp/AwsXrayContextDataProvider.java>

**Comment by advanderveer** [comment]
@mderriey thank you for chiming in! The links are usefull, I'll need to investigate a bit alter but will then report if it worked for me or not.

**Comment by mderriey** [comment]
Friendly bump on this 😉

**Comment by advanderveer** [comment]
@mderriey sorry, it looks like the deployment that had this issue is not online anymore. So it's not easy for me to check this at this point. I will try to keep it in mind as we're adding tracing to other projects and inform you at that point.

**Comment by kyle-query** [comment]
I've followed the steps in the thread and have a different result, but not a success. The traces look like:

```json
{
    "Id": "1-6423651e-bf58f537036ef0ef2c2b6b44",
    "Duration": 6.885,
    "LimitExceeded": false,
    "Segments": [
        {
            "Id": "339f39ddf5531aa3",
            "Document": {
                "id": "339f39ddf5531aa3",
                "name": "dev-query-engine",
                "start_time": 1680041246.7139075,
                "trace_id": "1-6423651e-bf58f537036ef0ef2c2b6b44",
                "end_time": 1680041251.1205487,
                "parent_id": "b81543f1e4b84da1",
                "aws": {
                    "account_id": "...",
                    "ecs": {
                        "...": "..."
                    },
                    "xray": {
                        "auto_instrumentation": false,
                        "sdk_version": "1.16.0",
                        "sdk": "opentelemetry for python"
                    },
                    "cloudwatch_logs": [
                        {
                            "log_group": "dev-query-engine"
                        }
                    ]
                },
                "metadata": {
                    "default": {
```

Opening this trace in CloudWatch shows segments from all the services involved, but once we started exporting the `aws.cloudwatch_logs` attribute, the Logs section of the UI shows only "An error occurred when fetching your data". Now the logs from other services that used to appear in the trace no longer are shown either.

Clicking "View in CloudWatch Insights" now has no log groups preselected, where it used to have several groups pre-selected. Manually selecting the relevant log groups does indeed show application logs from all services, including the one instrumented with OTEL.

I don't think I've seen anyone report that this feature works, so maybe someone will chime in to let us know if it does work under some circumstances.

**Comment by mderriey** [comment]
Hey @kyle-query 

I also tried doing it with just the log group name, and got thew same outcome as you did.

For us, the only thing that worked was having the `cloudwatch_logs` section as below:

```json
"aws": {
  "cloudwatch_logs": [
    {
      "log_group": "<log-group-name>",
      "arn": "arn:aws:logs:<region>:<account-id>:log-group:<log-group-name>"
    }
  ]
},
```

Please note that the `arn` property cannot end with `:*`, see https://github.com/aws-observability/aws-otel-collector/issues/1865

**Comment by kyle-query** [comment]
Thanks @mderriey, we have it working now!

**Comment by willarmiros** [comment]
Hi folks, I just closed #1865 since the CloudWatch Console has fixed the `:*` issue!

> The CloudWatch Console team has fixed this issue on our end, so the correlation should work with or without the `:*` suffix! Please let us know if there are still issues, but I'll close this for now!

I think this issue can be closed as well, since it seemed that bug was the one primarily causing pain here. In addition, we made another improvement I mentioned above:

> I think the action should be on us to update the X-Ray exporter. Technically the spec requires aws.log.group.names to be a slice, but we're seeing an increased number of use cases where setting it as string is just more sensible, so the X-Ray exporter should translate both.

in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16859. @mderriey @nikhil-pandey are you ok with this being closed now?

**Comment by rapphil** [comment]
Closing as this was solved.
